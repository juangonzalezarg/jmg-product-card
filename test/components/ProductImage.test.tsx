import * as React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/productos';

describe('ProductImage', () => {
    test('debe de mostrar el componente sin imagen del producto', () => {

        const wrapper = renderer.create(
            <ProductImage img="https://hola.jpg"/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    test('debe de mostrar el componente con imagen del producto', () => { 
        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage/>
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
     })
});
