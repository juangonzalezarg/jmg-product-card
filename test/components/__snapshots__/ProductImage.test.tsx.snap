// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente sin imagen del producto 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
